### ISSUE TITLE
Update for WordPress 5.3

### ISSUE THREAD

**Issue by kayvanaarssen** [LuRenJiasWorld/WP-Editor.md#380]
Hi,

WordPress 5.3 is out. But your plugin does now not work anymore.

Do you have a fix available?

**Comment by mingmingsuper** [comment]
Hi, I  have the problem too. 
Hope you update a new version for WordPress 5.3, Thanks!

**Comment by Skywt2003** [comment]
我也有一样的问题，后端编辑器无法显示，但是前端 Markdown 语法渲染正常。

**Comment by LuRenJiasWorld** [comment]
你可以先使用wp-downgrade插件降级到WordPress 5.2版本，然后静候开发者更新。

根据报错信息分析，可能是因为WordPress5.3不再暴露jQuery的`$`函数。

You can downgrade to WordPress 5.2 using the wp-downgrade plugin, and wait for developers to update the fix for this editor.

It seems like the newest WordPress 5.3 does not expose the `$` function of jQuery anymore, which is used by this editor.

**Comment by kayvanaarssen** [comment]
> 你可以先使用wp-downgrade插件降级到WordPress 5.2版本，然后静候开发者更新。
> 
> 根据报错信息分析，可能是因为WordPress5.3不再暴露jQuery的`$`函数。
> 
> You can downgrade to WordPress 5.2 using the wp-downgrade plugin, and wait for developers to update the fix for this editor.
> 
> It seems like the newest WordPress 5.3 does not expose the `$` function of jQuery anymore, which is used by this editor.

Allready downgraded for now, but would be good to have an update for the module 👍

**Comment by LuRenJiasWorld** [comment]
> > 你可以先使用wp-downgrade插件降级到WordPress 5.2版本，然后静候开发者更新。
> > 根据报错信息分析，可能是因为WordPress5.3不再暴露jQuery的`$`函数。
> > You can downgrade to WordPress 5.2 using the wp-downgrade plugin, and wait for developers to update the fix for this editor.
> > It seems like the newest WordPress 5.3 does not expose the `$` function of jQuery anymore, which is used by this editor.
> 
> Allready downgraded for now, but would be good to have an update for the module 👍

Also, you can add a JavaScript code in the <head> part, before any JavaScript code or link, the code is shown below:
```javascript
<script>
var $ = jQuery;
</script>
```
You can use the `wp_enqueue_script()` function in WordPress to inject this code into all pages.

https://wordpress.org/support/topic/jquery-error-in-custom-theme-after-upgrading-to-5-3/
Reference from this link.

**Comment by LuRenJiasWorld** [comment]
> > 你可以先使用wp-downgrade插件降级到WordPress 5.2版本，然后静候开发者更新。
> > 根据报错信息分析，可能是因为WordPress5.3不再暴露jQuery的`$`函数。
> > You can downgrade to WordPress 5.2 using the wp-downgrade plugin, and wait for developers to update the fix for this editor.
> > It seems like the newest WordPress 5.3 does not expose the `$` function of jQuery anymore, which is used by this editor.
> 
> Allready downgraded for now, but would be good to have an update for the module 👍

In addition, I developed a plugin to solve this problem, this plugin is also valid for other existed plugins compatibility issues.

The GitHub Repo of this plugin is https://github.com/LuRenJiasWorld/jQuery-WP53-Fix

You can test it out and feel free if you have any questions or encountering any issues.

**Comment by kayvanaarssen** [comment]
> > > 你可以先使用wp-downgrade插件降级到WordPress 5.2版本，然后静候开发者更新。
> > > 根据报错信息分析，可能是因为WordPress5.3不再暴露jQuery的`$`函数。
> > > You can downgrade to WordPress 5.2 using the wp-downgrade plugin, and wait for developers to update the fix for this editor.
> > > It seems like the newest WordPress 5.3 does not expose the `$` function of jQuery anymore, which is used by this editor.
> > 
> > 
> > Allready downgraded for now, but would be good to have an update for the module 👍
> 
> In addition, I developed a plugin to solve this problem, this plugin is also work for other existed plugin's compatibility issues.
> 
> The GitHub Repo of this plugin is https://github.com/LuRenJiasWorld/jQuery-WP53-Fix
> 
> You can test it out and feel free if you have any questions or encountering any issues.

Nice, for now. But stll the original plugin needs to be updated

**Comment by LuRenJiasWorld** [comment]
> Nice, for now. But stll the original plugin needs to be updated

Sure, let me analyze the code now and see how to solve this problem fundamentally.

**Comment by sharpbai** [comment]
可以暂时编辑`src/Utils/Settings.php`，把第91行
```
jQuery( function() { $("#editor_mermaid\\\\[mermaid_config\\\\]") ...
```
里面的`$`替换成`jQuery`

```
jQuery( function() { jQuery("#editor_mermaid\\\\[mermaid_config\\\\]") ...
```
问题即可以解决。

**Comment by LuRenJiasWorld** [comment]
> 可以暂时编辑`src/Utils/Settings.php`，把第91行
> You can modify `src/Utils/Settings.php` for the moment. In line 91
> 
> ```
> jQuery( function() { $("#editor_mermaid\\\\[mermaid_config\\\\]") ...
> ```
> 
> 里面的`$`替换成`jQuery`
> swap `$` with `jQuery`
> 
> ```
> jQuery( function() { jQuery("#editor_mermaid\\\\[mermaid_config\\\\]") ...
> ```
> 
> 问题即可以解决。
> to solve the problem.

但是似乎这样并不能解决问题，就算是在Settings.php中，第671行左右也还是有对`$`的引用。如果是逐个替换，感觉工作量会很大。

**Comment by LuRenJiasWorld** [comment]
Hello, everyone. Thanks for your contribution and this issue has been solved in the latest version(10.0.8). Please update the plugin in your WordPress admin page and see if it works fine. If you have any other issues, please reply.

各位好，感谢大家的反馈，本插件已经发布了最新版本（10.0.8）来修复此问题，请更新到最新版本即可解决。如果遇到其他问题，欢迎继续回复本Issue，我将会跟进解决。
